tests/functional/report_viewer_api/test_get_lines_in_file.py

```diff
@@ -0,0 +1,109 @@
+import logging
+import os
+import unittest
+
+from libtest import env
+
+from codeCheckerDBAccess_v6.ttypes import Encoding
+from codeCheckerDBAccess_v6.ttypes import LinesInFilesRequested
+from codeCheckerDBAccess_v6.ttypes import ReportFilter
+
+class TestRunFilter(unittest.TestCase):
+
+    _ccClient = None
+
+    def setUp(self):
+        test_workspace = os.environ['TEST_WORKSPACE']
+
+        test_class = self.__class__.__name__
+        print('Running ' + test_class + ' tests in ' + test_workspace)
+
+        # Get the clang version which is tested.
+        self._clang_to_test = env.clang_to_test()
+
+        self._testproject_data = env.setup_test_proj_cfg(test_workspace)
+        self.assertIsNotNone(self._testproject_data)
+
+        self._cc_client = env.setup_viewer_client(test_workspace)
+        self.assertIsNotNone(self._cc_client)
+
+        # Get the run names which belong to this test.
+        run_names = env.get_run_names(test_workspace)
+
+        runs = self._cc_client.getRunData(None)
+
+        test_runs = [run for run in runs if run.name in run_names]
+
+        self._runid = test_runs[0].runId
+
+    def __get_file_id_for_path(self, run_id, file_path):
+        # Returns file id for the given file path in the given run.
+        file_filter = ReportFilter(filepath=[file_path])
+        run_results = self._cc_client.getRunResults(
+            [run_id], 1, 0, None, file_filter, None)
+
+        self.assertTrue(len(run_results))
+        return run_results[0].fileId
+
+    def __check_lines_in_source_file_contents(self, lines_in_files_requested,
+                                              encoding, expected):
+        """
+        Get line content information with the given encoding and check the
+        results.
+        """
+        file_to_lines_map = self._cc_client.getLinesInSourceFileContents(
+            lines_in_files_requested, encoding)
+        for file_id in expected:
+            for line in expected[file_id]:
+                source_line = file_to_lines_map[file_id][line]
+                if encoding == Encoding.BASE64:
+                    source_line = base64.b64decode(source_line)
+
+                self.assertEqual(source_line,
+                                 expected[file_id][line])
+
+    def test_get_lines_in_source_file_contents(self):
+        """
+        Get line content information for multiple files in different positions.
+        """
+        runid = self._runid
+        logging.debug('Get line content information from the db for runid: ' +
+                      str(runid))
+
+        # Get reports by file to get a file id.
+        file_id_1 = self.__get_file_id_for_path(runid, "*call_and_message.cpp")
+        file_id_2 = self.__get_file_id_for_path(runid, "*divide_zero.cpp")
+
+        lines_in_files_requested = [
+            LinesInFilesRequested(fileId=file_id_1, lines=[15, 20]),
+            LinesInFilesRequested(fileId=file_id_2, lines=[12, 19])]
+
+        expected = {
```

Comment:
I feel this here is an extra maintenance burden. We know where the stored files reside, so in case the source files for the tests change, the `expected` dict should be built via reading the *actual* source files on the disk, and not via coding in parts of the source file here too.